lib/wallaroo/core/windows/panes_range_windows.pony

```diff
@@ -268,6 +274,11 @@ class _PanesSlidingWindows[In: Any val, Out: Any val, Acc: State ref] is
       working_idx = (working_idx + 1) % old_panes_size
     end
 
+    ifdef debug then
+      // We should never create an unreasonable amount of frames
+      // this particular number is up for debate.
+      Invariant(new_pane_count < 1000)
```

Comment:
`frames` -> `panes`

We're not enforcing this yet, so I wouldn't call this an invariant.  What if we get a message with a timestamp way in the future?  That's a valid possible property of the input stream that we don't yet handle well (we create an excessive number of panes).  If we don't already have an issue, we need one for shrinking the number of panes dynamically and potentially buffering and special-casing messages whose timestamps are too far in the future (though we'd need to do this efficiently).